### ISSUE TITLE
Private group conversations duplication with unreadable messages

### ISSUE THREAD

**Issue by tcottier-enalean** [mattermost/mattermost-server#12168]
#### Summary

Some private group conversation created between 3 or more users are duplicated and make part of the old conversation unreadable

#### Steps to reproduce
Create a private conversation between you and 2 others users, chat a bit then change channel. 

Sometimes when someone sends a message to this channel and you try to reenter the conversation another conversation opens with the same persons, and you can not access the new one. The conversation menu on the left shows 2 conversations but one is unreachable

![bug-mattermost](https://user-images.githubusercontent.com/8102478/64707581-7d6c6100-d4b3-11e9-8a9d-ff8f3b8e406f.gif)

We are running Mattermost Version: 5.14.2

#### Expected behavior
We only have one private group conversation with other people and we can see all the messages.

#### Observed behavior (that appears unintentional) 

I have the following messages in my logs about these particular channels (anonymized):
```
{"level":"error","ts":1568208095.8175354,"caller":"mlog/log.go:173","msg":"Unable to find the existing channel","path":"/api/v4/teams/name/mycompany/channels/name/68f6fd8c7e5397690ff1c300b4aa6c9db0af7806","request_id":"uc6w5netu7dmufh6naoe81z4gy","ip_addr":"x.x.x.x","user_id":"mrqojzwwp7nk7xi714mzf7fh7c","method":"GET","err_where":"SqlChannelStore.GetByName","http_code":400,"err_details":"teamId=do4h39o4dfbptf4tq8kgzyb8ay, name=68f6fd8c7e5397690ff1c300b4aa6c9db0af7806, gorp: multiple rows returned for: SELECT * FROM Channels WHERE (TeamId = :TeamId OR TeamId = '') AND Name = :Name AND DeleteAt = 0 - [map[Name:68f6fd8c7e5397690ff1c300b4aa6c9db0af7806 TeamId:do4h39o4dfbptf4tq8kgzyb8ay]]"}
```

I can also see multiple conversations that have a duplicate name in  the Channels table:

```
mattermost=# select name, teamid, type, count(*) from Channels group by name, teamid, type  having count(*) > 1;
                   name                   | teamid | type | count 
------------------------------------------+--------+------+-------
 8a65b60bb6943bd17b66258d3e384d7aac2a5e17 |        | G    |     3
 e1a76d2485f745ffaac7bda1623fdf03c211148c |        | G    |     2
 131d27b5b8aaa4d42c8241b1717d0e57548fd3ed |        | G    |     2
 331fb98016a8933e1ec3b373809b6ea413c4f75f |        | G    |     2
 4de7ad0ff6975f4f02fc954ae41f9016212f36af |        | G    |     2
 632df62c9b7f76f09ffe30e8d4719aefe5e56e72 |        | G    |     2
 3b2790bc654f05ab6656cafab478c21a4ec04ed0 |        | G    |     2
 50ca16daa8b7b703b53eef56b383e10e7dea8669 |        | G    |     3
 8415fe4885becf0fbd45ddcfe45daf6337e51ac3 |        | G    |     2
 1a83cc5e9d86db7e9a17674af527906258f03e55 |        | G    |     4
 d909891c3dc2c5e953697b4e5f511dc29a7b41ba |        | G    |     2
 852fbfdfbc730c9ee9d2f758cfc1ffb54b91ff9c |        | G    |     2
 c8005fd8ca41add6063360cded8f467559e20da0 |        | G    |     4
 bfb8dede84fa959ea2ea4dabe2f1f41a8fb4450e |        | G    |     4
 23223c0f9df93f70c45b6f8f5e0deb5b1cc57cbb |        | G    |     2
 6d27fcd4f1774b292693c5d0330eb8a39cf90b78 |        | G    |     2
 f7711c04d442e5f228d6a24c22caba84fccef7a9 |        | G    |     2
 babc1e759ec154a248c29a2d56b5740ecbcc4c93 |        | G    |     2
 2c049890c160cbd57f39fac052e643ba14e77de5 |        | G    |     2
 931140139135fc94a0fe5f3513cab56f10f539aa |        | G    |     2
 2cf586ba48766064f42874805d7fb4d7c14a5a85 |        | G    |     2
 4dfc2a8f26d1172d6d73f586a03709c6eead028f |        | G    |     2
 fa411e7618897f015d8fb70cd1e677ae7d3eef0e |        | G    |     5
(23 rows)
```

An example of a duplicate record (anonymized):

```
mattermost=# SELECT name, id, displayname, totalmsgcount, type, createat, updateat FROM Channels where name like 'fa411e7618897f015d8fb70cd1e677ae7d3eef0e';
                   name                   |             id             |       displayname       | totalmsgcount | type |   createat    |   updateat    
------------------------------------------+----------------------------+-------------------------+---------------+------+---------------+---------------
 fa411e7618897f015d8fb70cd1e677ae7d3eef0e | 4hzy38s9yp8qdf58mdikpnyphr | user1, user2, user3 |            12 | G    | 1531821343153 | 1558444775355
 fa411e7618897f015d8fb70cd1e677ae7d3eef0e | 3114yjksyfn85x71rsrg5bthqw | user1, user2, user3 |             0 | G    | 1558444284941 | 1558444284941
 fa411e7618897f015d8fb70cd1e677ae7d3eef0e | qtr4op71r3ggmpqymfofef5bko | user1, user2, user3 |             0 | G    | 1558444397630 | 1558444397630
 fa411e7618897f015d8fb70cd1e677ae7d3eef0e | nu3bn8am3jr438p61pbqhuwy5y | user1, user2, user3 |             0 | G    | 1558444432484 | 1558444432484
 fa411e7618897f015d8fb70cd1e677ae7d3eef0e | 7bnq5xsahtnpum6xykzsegp3sw | user1, user2, user3 |             0 | G    | 1558443420659 | 1558443420659
(5 rows)
```

I'm not sure to be fully clear on this issue as this is not really easy to explain what happens. I can run any test you want on my server if it can help.

Thanks for your help.

**Comment by amyblais** [comment]
Hi @tcottier-enalean, Would you be open to reviewing this page and sending us any relevant additional information about your environment: https://docs.mattermost.com/process/support.html#general-questions-for-any-issues?

Have you seen any patterns on when this occurs and doesn't occur?

**Comment by tcottier-enalean** [comment]
Hi, sorry I did not see this page.

First of all no we did not see any patterns except it occurs only in private group conversations.
We are running Mattermost on docker (on a CentOS 7 dedicated serv), using the official docker compose to build and run the images. We have a Postgresql database.

Mattermost version is the latest stable CE: 5.14.2
We tested on the desktop app (on macOS), Chrome and Firefox and it makes no differences.
